cpp/tests/binaryop/binop-compiled-test.cpp

```diff
@@ -557,7 +557,11 @@ auto NullOp_Result(cudf::column_view lhs, cudf::column_view rhs)
   std::transform(thrust::make_counting_iterator(0),
                  thrust::make_counting_iterator(lhs.size()),
                  result.begin(),
-                 [&lhs_data, &lhs_mask, &rhs_data, &rhs_mask, &result_mask](auto i) -> TypeOut {
+                 [&lhs_data    = lhs_data,
+                  &lhs_mask    = lhs_mask,
+                  &rhs_data    = rhs_data,
+                  &rhs_mask    = rhs_mask,
+                  &result_mask = result_mask](auto i) -> TypeOut {
```

Comment:
I'm not sure I've understood the need for this either.  Why is repeating the variable name here preferred?

Reply:
Structure binding cannot be captured automatically so we have to use this trick.

Reply:
> Structure binding cannot be captured automatically...

In which case then, how was it working before?  The definitions of `lhs_data` etc. haven't changed, right?

Reply:
We were getting lucky because gcc implements this as an extension even when only compiling under C++17, but other compilers will not be so friendly: https://godbolt.org/z/9fnq6Ex84.